src/google/protobuf/repeated_ptr_field.h

```diff
@@ -693,11 +693,19 @@ class PROTOBUF_EXPORT RepeatedPtrFieldBase {
   Rep* rep_;
 
   template <typename TypeHandler>
-  static inline typename TypeHandler::Type* cast(void* element) {
+  #ifdef _MSC_BUILD 
+  static inline typename TypeHandler::Type* (cast) (void* element){
```

Comment:
This makes me suspect some kind of rude macro is getting pushed into our namespace. Do you know where it comes from? I don't think `windows.h` has a `cast` macro. We should figure out what's causing this so we can fix this globally in `port_def.inc`.

Reply:
Thanks for your patience, it derives from my initial problem which is like when I include ```repeated_ptr_field``` will cause compiler error, changing the template declaration like above will solve this problem and successfully build. I m not sure what will happen on other platforms.

Reply:
Does it work to `#undef cast` in the list of `push_macro` calls in https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/port_def.inc ?

Also add pop_macro in: https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/port_undef.inc